libraries/botbuilder-dialogs-adaptive/src/actions/sendHandoffActivity.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+
+import { Expression, ObjectExpression, ObjectExpressionConverter } from 'adaptive-expressions';
+import { EventFactory, ResourceResponse, Transcript } from 'botbuilder';
+
+import {
+    Converter,
+    ConverterFactory,
+    Dialog,
+    DialogConfiguration,
+    DialogContext,
+    DialogTurnResult,
+} from 'botbuilder-dialogs';
+
+export interface SendHandoffActivityConfiguration extends DialogConfiguration {
+    context?: string | Expression | ObjectExpression<any>;
+    transcript?: string | Expression | ObjectExpression<any>;
+}
+
+/**
+ * Sends a handoff activity.
+ */
+export class SendHandoffActivity extends Dialog implements SendHandoffActivityConfiguration {
+    public static $kind = 'Microsoft.SendHandoffActivity';
+
+    /**
+     * Context to send with handoff activity
+     */
+    public context: ObjectExpression<Record<string, unknown>>;
+
+    /**
+     * Transcript to send with handoff activity
+     */
+    public transcript: ObjectExpression<Transcript>;
+
+    public getConverter(property: keyof SendHandoffActivityConfiguration): Converter | ConverterFactory {
+        switch (property) {
+            case 'context':
+                return new ObjectExpressionConverter<Record<string, unknown>>();
+            case 'transcript':
+                return new ObjectExpressionConverter<Transcript>();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Starts a new [Dialog](xref:botbuilder-dialogs.Dialog) and pushes it onto the dialog stack.
+     *
+     * @param {DialogContext} dc dialog context
+     * @returns {Promise<DialogTurnResult<ResourceResponse>>} result of sending handoff activity
+     */
+    async beginDialog(dc: DialogContext): Promise<DialogTurnResult<ResourceResponse>> {
+        const context = this.context?.getValue(dc.state);
+        const transcript = this.transcript?.getValue(dc.state);
+
+        const activity = EventFactory.createHandoffInitiation(dc.context, context, transcript);
+        const response = await dc.context.sendActivity(activity);
+
+        return dc.endDialog(response);
```

Comment:
Why is this unsettling?

Reply:
To me a dialog is a multiple-interaction process, not something that begins and ends immediately. Something that meets the latter scenario could be said to be a single-turn interaction and single-turn interactions do not need Dialogs. By adding Dialogs, the application becomes more fragile through the introduction of unnecessary moving parts (e.g. state storage, additional libraries).

However, Composer blurs the lines to build a seamless UX for customers and the "Dialog" in Composer is just a way to organize code. The generated bot runtime via Composer is intrinsically tied to Dialogs (Adaptive or otherwise), so the lens to review the PR with is different from the one I initially used. I now think it's fine as is.